nwapp/utils/notifier.js

```diff
@@ -28,14 +36,109 @@ function playNotificationSound() {
   audio = null;
 }
 
-module.exports = function (spec) {
-  if (!settings.showNotifications) return;
-  playNotificationSound();
 
-  var notification = new window.Notification(spec.title, {
-    body: spec.message,
-    icon: getNotificationImageForUri(spec.link)
+// Cache lookup
+var urlFilePathCacheMap = new Map();
+var tempAvatarDirectory = path.join(os.tmpDir(), 'gitter-notification-avatars_');
+log.info('Saving notification avatars at: ' + tempAvatarDirectory);
+var ensureTempAvatarDirectory = ensureDir(tempAvatarDirectory)
+  .then(function() {
+    return tempAvatarDirectory;
+  });
+
+events.on('app:quit', function() {
+  log.info('App is quiting, cleanup avatars');
+  try {
+    rimraf.sync(tempAvatarDirectory);
+  }
+  catch(err) {
+    log.error('Something went wrong while cleaning up the avatars: ' + err.message + err.stack);
+  }
+});
+
+function downloadAndCache(url) {
+  return new Promise(function(resolve, reject) {
+    var tempFileStream = temp.createWriteStream({
+      dir: tempAvatarDirectory
+    });
+
+    tempFileStream
+      .on('finish', function() {
+        var tempFilePath = tempFileStream.path;
+        // Store it on our map as a cache lookup
+        urlFilePathCacheMap.set(url, tempFilePath);
+        resolve(tempFilePath);
+      })
+      .on('error', function(err) {
+        reject(err);
+      });
+
+    request({
+        url: url,
+        method: 'GET',
+        timeout: 750
+      })
+      .on('error', function(err) {
+        reject(err);
+      })
+      .pipe(tempFileStream);
   });
+}
+function getFileUrlForHttpUrl(url) {
+  return ensureTempAvatarDirectory.then(function() {
+      var cachedFilePath = urlFilePathCacheMap.get(url);
+      return cachedFilePath || downloadAndCache(url);
+    })
+    .then(function(filePath) {
+      return 'file://' + filePath;
+    });
+}
+
+
+var notifierDefaults = {
+  title: '',
+  message: '',
+  // gitterHQ logo
+  icon: 'https://avatars0.githubusercontent.com/u/5990364?v=3&s=60',
```

Comment:
From my testing, this doesn't seem to happen. It looks like it might for Mac if we made a custom build of `node-notifier` and added our app icon.

OSX:
![](https://i.imgur.com/ZOSgdpA.png)
Windows:
![](http://i.imgur.com/HUBKgKN.png)
Ubuntu:
![](https://i.imgur.com/tFsLVnk.png)